hathor/merged_mining/bitcoin.py

```diff
@@ -38,7 +52,7 @@ def __bytes__(self) -> bytes:
         ])
 
     @classmethod
-    def from_dict(cls, params: dict) -> 'BitcoinBlockHeader':
+    def from_dict(cls, params: Dict) -> 'BitcoinBlockHeader':
```

Comment:
`Dict[str, Any]`?

Reply:
I honestly can't be bothered using `Dict[str, Any]`, it isn't that much better than `Dict[Any, Any]` which is just `Dict` which is much less annoying to type. I wish there as an easy way to describe a `Dict` that can be json de/serialized. [But this isn't quite possible yet.](https://github.com/python/typing/issues/182). But we could use Guido's suggestion of doing `JsonDict = Dict[str, Any]` add that to `utils.py` and use where it makes sense (most uses of `Dict` probably).

Reply:
I agree that they are quite the same. But using `Dict[str, Any]` is a step towards strict mode verification by mypy. I like the idea of using a `JsonDict`.

Reply:
I'll do this on a new PR.

Reply:
I've made some progress on `refactor/typing-json-dict`. Trying to be a little more specific than `Dict[str, Any]` has been taking far more work than I'd like to put in now. I'll come back to it later because I may have spotted a few issues because of this. But I won't include any of it on this PR, since there isn't any benefit to the issue it solves.

Reply:
The draft PR is here: https://github.com/HathorNetwork/hathor-core/pull/126 I won't be putting more effort on this for now.